crates/providers/providers-alloy/src/beacon_client.rs

```diff
@@ -193,7 +204,7 @@ impl BeaconClient for OnlineBeaconClient {
             kona_macros::inc!(gauge, Metrics::BEACON_CLIENT_ERRORS, "method" => "spec");
         }
 
-        result
+        result.map_err(|err| BeaconClientError::Http(err))
```

Comment:
nit, you could also use (for forward compatibility):
```suggestion
        Ok(result?)
```